Debian 9

Hi, thanks for your work. 

Is there plan to support Debian 9 ?

Thanks.

Hello,

I tried to install it on a fresh debian 9.

To be able to start the process, you have to make these changes in the script file:
```
403 if [ "${DIST}" = "Debian" ]; then
404   DIST_PKGS="${DIST_PKGS} openjdk-8-jdk"
405 fi
```
Debian 8 -> openjdk-7-jdk
Debian 9 -> openjdk-8-jdk

And you have to install manually 2 packages from debian wheezy because they don't exist anymore in stretch: 
```
libpng12-0 -> https://packages.debian.org/wheezy/libpng12-0
libpng12-dev -> https://packages.debian.org/wheezy/libpng12-dev
```
Why wheezy and not jessie: https://unix.stackexchange.com/questions/331803/cannot-install-libpng12-0-on-debian.

And you have to remove those 2 packages from the apt-get install command on line 407 (and libpng12-dev must be removed from line 408 as well.. it's 2 times there.. that's a small error)

After that, I could start the script.

The Lo compilation finished ok, but then, the script failed on the poco stuff.

Here are the last lines with the error (the Crypto section):
```
make -C /opt/poco-1.7.8p3-all/Crypto
make[1] : on entre dans le répertoire « /opt/poco-1.7.8p3-all/Crypto »
** Creating dependency info for src/OpenSSLInitializer.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/OpenSSLInitializer.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/OpenSSLInitializer.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/X509Certificate.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/X509Certificate.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/X509Certificate.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/DigestEngine.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/DigestEngine.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/DigestEngine.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/RSADigestEngine.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/RSADigestEngine.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/RSADigestEngine.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/RSAKeyImpl.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/RSAKeyImpl.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/RSAKeyImpl.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/RSAKey.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/RSAKey.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/RSAKey.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/RSACipherImpl.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/RSACipherImpl.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/RSACipherImpl.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.
7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/Cpp
Unit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/
Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite
/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/
include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_
REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/CryptoTransform.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/CryptoTransform.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/CryptoTransform.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poc
o-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all
/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-
all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQ
Lite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompi
ler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600
 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/CryptoStream.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/CryptoStream.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/CryptoStream.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.
8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUn
it/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Ut
il/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/i
nclude -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/in
clude -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_RE
ENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/CipherKeyImpl.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/CipherKeyImpl.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/CipherKeyImpl.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.
7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/Cpp
Unit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/
Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite
/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/
include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_
REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/CipherKey.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/CipherKey.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/CipherKey.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-al
l/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/inc
lude -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/inc
lude -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include
 -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include 
-I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRAN
T -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/CipherImpl.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/CipherImpl.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/CipherImpl.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-
all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/i
nclude -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/CipherFactory.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/CipherFactory.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/CipherFactory.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
** Creating dependency info for src/Cipher.cpp
mkdir -p /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64
/opt/poco-1.7.8p3-all/build/script/makedepend.gcc  src/Cipher.cpp /opt/poco-1.7.8p3-all/Crypto/.dep/Linux/x86_64/Cipher.d /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
mkdir -p /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_static
mkdir -p /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_static
mkdir -p /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/release_shared
mkdir -p /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared
** Compiling src/Cipher.cpp (debug, shared)
g++  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL  -g -D_DEBUG -fPIC -c src/Cipher.cpp -o /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared/Cipher.o
** Compiling src/CipherFactory.cpp (debug, shared)
g++  -Iinclude -I/opt/poco-1.7.8p3-all/CppUnit/include -I/opt/poco-1.7.8p3-all/CppUnit/WinTestRunner/include -I/opt/poco-1.7.8p3-all/Foundation/include -I/opt/poco-1.7.8p3-all/XML/include -I/opt/poco-1.7.8p3-all/JSON/include -I/opt/poco-1.7.8p3-all/Util/include -I/opt/poco-1.7.8p3-all/Net/include -I/opt/poco-1.7.8p3-all/Crypto/include -I/opt/poco-1.7.8p3-all/NetSSL_OpenSSL/include -I/opt/poco-1.7.8p3-all/Data/include -I/opt/poco-1.7.8p3-all/Data/SQLite/include -I/opt/poco-1.7.8p3-all/Data/ODBC/include -I/opt/poco-1.7.8p3-all/Data/MySQL/include -I/opt/poco-1.7.8p3-all/MongoDB/include -I/opt/poco-1.7.8p3-all/Zip/include -I/opt/poco-1.7.8p3-all/PageCompiler/include -I/opt/poco-1.7.8p3-all/PageCompiler/File2Page/include -Wall -Wno-sign-compare -DPOCO_BUILD_HOST='"'lool'"' -DPOCO_TARGET_OSNAME='"'Linux'"' -DPOCO_TARGET_OSARCH='"'x86_64'"'  -D_XOPEN_SOURCE=600 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL  -g -D_DEBUG -fPIC -c src/CipherFactory.cpp -o /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared/CipherFactory.o
In file included from include/Poco/Crypto/CipherKeyImpl.h:24:0,
                 from include/Poco/Crypto/CipherKey.h:24,
                 from src/CipherFactory.cpp:19:
include/Poco/Crypto/OpenSSLInitializer.h:26:28: fatal error: openssl/crypto.h: Aucun fichier ou dossier de ce type
 #include <openssl/crypto.h>
                            ^
compilation terminated.
/opt/poco-1.7.8p3-all/build/rules/compile:53 : la recette pour la cible « /opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared/CipherFactory.o » a échouée
make[1] : on quitte le répertoire « /opt/poco-1.7.8p3-all/Crypto »
make[1]: *** [/opt/poco-1.7.8p3-all/Crypto/obj/Linux/x86_64/debug_shared/CipherFactory.o] Erreur 1
make: *** [Crypto-libexec] Erreur 2
Makefile:201 : la recette pour la cible « Crypto-libexec » a échouée
```

Any idea how to fix that?

It failed.
Another error linked to ssl it seems.

This one occurred on this command line in lib/onlineBuild.sh (line 39):
```
sudo -Hu lool make -j$cpu --directory=${lool_dir}
```
And here are the errors:
```
make : on entre dans le répertoire « /opt/online »
make  all-recursive
make[1] : on entre dans le répertoire « /opt/online »
Making all in .
make[2] : on entre dans le répertoire « /opt/online »
  CXX      net/Ssl.o
net/Ssl.cpp: In constructor ‘SslContext::SslContext(const string&, const string&, const string&)’:
net/Ssl.cpp:46:5: error: ‘void OPENSSL_config(const char*)’ is deprecated [-Werror=deprecated-declarations]
     OPENSSL_config(nullptr);
     ^~~~~~~~~~~~~~
In file included from /usr/include/openssl/ct.h:13:0,
                 from /usr/include/openssl/ssl.h:61,
                 from net/Ssl.hpp:19,
                 from net/Ssl.cpp:13:
/usr/include/openssl/conf.h:92:1: note: declared here
 DEPRECATEDIN_1_1_0(void OPENSSL_config(const char *config_name))
 ^
net/Ssl.cpp:46:27: error: ‘void OPENSSL_config(const char*)’ is deprecated [-Werror=deprecated-declarations]
     OPENSSL_config(nullptr);
                           ^
In file included from /usr/include/openssl/ct.h:13:0,
                 from /usr/include/openssl/ssl.h:61,
                 from net/Ssl.hpp:19,
                 from net/Ssl.cpp:13:
/usr/include/openssl/conf.h:92:1: note: declared here
 DEPRECATEDIN_1_1_0(void OPENSSL_config(const char *config_name))
 ^
net/Ssl.cpp: In member function ‘void SslContext::initDH()’:
net/Ssl.cpp:236:7: error: invalid use of incomplete type ‘DH {aka struct dh_st}’
     dh->p = BN_bin2bn(dh1024_p, sizeof(dh1024_p), 0);
       ^~
In file included from /usr/include/openssl/crypto.h:31:0,
                 from /usr/include/openssl/comp.h:16,
                 from /usr/include/openssl/ssl.h:47,
                 from net/Ssl.hpp:19,
                 from net/Ssl.cpp:13:
/usr/include/openssl/ossl_typ.h:104:16: note: forward declaration of ‘DH {aka struct dh_st}’
 typedef struct dh_st DH;
                ^~~~~
net/Ssl.cpp:237:7: error: invalid use of incomplete type ‘DH {aka struct dh_st}’
     dh->g = BN_bin2bn(dh1024_g, sizeof(dh1024_g), 0);
       ^~
In file included from /usr/include/openssl/crypto.h:31:0,
                 from /usr/include/openssl/comp.h:16,
                 from /usr/include/openssl/ssl.h:47,
                 from net/Ssl.hpp:19,
                 from net/Ssl.cpp:13:
/usr/include/openssl/ossl_typ.h:104:16: note: forward declaration of ‘DH {aka struct dh_st}’
 typedef struct dh_st DH;
                ^~~~~
net/Ssl.cpp:238:7: error: invalid use of incomplete type ‘DH {aka struct dh_st}’
     dh->length = 160;
       ^~
In file included from /usr/include/openssl/crypto.h:31:0,
                 from /usr/include/openssl/comp.h:16,
                 from /usr/include/openssl/ssl.h:47,
                 from net/Ssl.hpp:19,
                 from net/Ssl.cpp:13:
/usr/include/openssl/ossl_typ.h:104:16: note: forward declaration of ‘DH {aka struct dh_st}’
 typedef struct dh_st DH;
                ^~~~~
net/Ssl.cpp:239:13: error: invalid use of incomplete type ‘DH {aka struct dh_st}’
     if ((!dh->p) || (!dh->g))
             ^~
In file included from /usr/include/openssl/crypto.h:31:0,
                 from /usr/include/openssl/comp.h:16,
                 from /usr/include/openssl/ssl.h:47,
                 from net/Ssl.hpp:19,
                 from net/Ssl.cpp:13:
/usr/include/openssl/ossl_typ.h:104:16: note: forward declaration of ‘DH {aka struct dh_st}’
 typedef struct dh_st DH;
                ^~~~~
net/Ssl.cpp:239:25: error: invalid use of incomplete type ‘DH {aka struct dh_st}’
     if ((!dh->p) || (!dh->g))
                         ^~
In file included from /usr/include/openssl/crypto.h:31:0,
                 from /usr/include/openssl/comp.h:16,
                 from /usr/include/openssl/ssl.h:47,
                 from net/Ssl.hpp:19,
                 from net/Ssl.cpp:13:
/usr/include/openssl/ossl_typ.h:104:16: note: forward declaration of ‘DH {aka struct dh_st}’
 typedef struct dh_st DH;
                ^~~~~
cc1plus: all warnings being treated as errors
Makefile:1165 : la recette pour la cible « net/Ssl.o » a échouée
make[2] : on quitte le répertoire « /opt/online »
make[2]: *** [net/Ssl.o] Erreur 1
make[1]: *** [all-recursive] Erreur 1
make: *** [all] Erreur 2
Makefile:1701 : la recette pour la cible « all-recursive » a échouée
make[1] : on quitte le répertoire « /opt/online »
Makefile:734 : la recette pour la cible « all » a échouée
make : on quitte le répertoire « /opt/online »
```

Here are the ssl packages installed on my server:
```
ii  libgnutls-openssl27:amd64 3.5.8-5+deb9u2 amd64        GNU TLS library - OpenSSL wrapper
ii  libssl-dev:amd64          1.1.0f-3       amd64        Secure Sockets Layer toolkit - development files
ii  libssl-doc                1.1.0f-3       all          Secure Sockets Layer toolkit - development documentation
ii  libssl1.0.2:amd64         1.0.2l-2       amd64        Secure Sockets Layer toolkit - shared libraries
ii  libssl1.1:amd64           1.1.0f-3       amd64        Secure Sockets Layer toolkit - shared libraries
ii  openssl                   1.1.0f-3       amd64        Secure Sockets Layer toolkit - cryptographic utility
```

These errors doesn't mean a lot for me. An old way of coding not good anymore??
The good point is that we are near the end of the compilation, just the onlineInstall.sh script and then the final test.

--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 
 loolwsd\.xml
+*officeonline-install

--- a/bin/systemSetup.sh
+++ b/bin/systemSetup.sh
@@ -46,7 +46,7 @@ if ! apt-get install ant sudo systemd wget zip make procps automake bison ccache
 flex g++ git gperf graphviz junit4 libcap-dev libcppunit-dev \
 libcppunit-doc libcunit1 libcunit1-dev libegl1-mesa-dev libfontconfig1-dev libgl1-mesa-dev \
 libgtk-3-dev libgtk2.0-dev libkrb5-dev libpcap0.8 libpcap0.8-dev libtool \
-libxml2-utils libxrandr-dev libxrender-dev libxslt1-dev libxt-dev m4 nasm openssl \
+libxml2-utils libxrandr-dev libxrender-dev libxslt1-dev libxt-dev m4 nasm openssl libssl-dev \
 pkg-config python-dev python-polib python3-dev uuid-runtime xsltproc \
   ${DIST_PKGS} -y; then
     exit 1

--- a/bin/systemSetup.sh
+++ b/bin/systemSetup.sh
@@ -43,7 +43,7 @@ if [ "${DIST}" = "Debian" ]; then
 fi
 
 if ! apt-get install ant sudo systemd wget zip make procps automake bison ccache \
-flex g++ git gperf graphviz junit4 libcap-dev libcppunit-dev \
+flex g++ git gperf graphviz junit4 libcap-dev libcppunit-dev build-essential \
 libcppunit-doc libcunit1 libcunit1-dev libegl1-mesa-dev libfontconfig1-dev libgl1-mesa-dev \
 libgtk-3-dev libgtk2.0-dev libkrb5-dev libpcap0.8 libpcap0.8-dev libtool \
 libxml2-utils libxrandr-dev libxrender-dev libxslt1-dev libxt-dev m4 nasm openssl libssl-dev \

--- a/bin/systemSetup.sh
+++ b/bin/systemSetup.sh
@@ -2,6 +2,31 @@
 # shellcheck disable=SC2154
 ###############################################################################
 ############################# System preparation ##############################
+ssl_fix_dirty(){
+  local apt_file="/etc/apt/sources.list"
+  local deb_="deb http://ftp.be.debian.org/debian/ jessie-backports main"
+  local deb_src="deb-src http://ftp.be.debian.org/debian/ jessie-backports main"
+  local check_deb=`grep "$deb_" $apt_file | wc -l`
+  local check_deb_src=`grep "$deb_src" $apt_file | wc -l`
+
+  if [ "$check_deb" == "0" ];then
+    echo $deb_ >> $apt_file
+  fi
+  if [ "$check_deb_src" == "0" ];then
+    echo $deb_src >> $apt_file
+  fi
+
+  (
+    echo "Package: openssl libssl1.0.0 libssl-dev libssl-doc"
+    echo "Pin: release a=jessie-backports"
+    echo "Pin-Priority: 1001"
+  ) | tee /etc/apt/preferences.d/00_ssl
+  apt-get update
+  if ! apt-get install openssl libssl-dev -y --allow-downgrades; then
+    exit 1
+  fi
+}
+
 if [ -n "${set_name:?}" ]; then
   echo "Searching for a set named $set_name..."
   my_set=$(FindOnlineSet "$set_name" "$lo_src_repo" "$set_core_regex" "$lool_src_repo" "$set_online_regex" "$set_version")
@@ -32,6 +57,7 @@ if [ "${DIST}" = "Ubuntu" ]; then
 fi
 if [ "${DIST}" = "Debian" ]; then
   if [ "${CODENAME}" = "stretch" ];then
+    ssl_fix_dirty
     DIST_PKGS="${DIST_PKGS} openjdk-8-jdk"
     DIST_PKGS="${DIST_PKGS} libpng16.16"
     DIST_PKGS="${DIST_PKGS} libpng-dev"

--- a/bin/systemSetup.sh
+++ b/bin/systemSetup.sh
@@ -47,7 +47,7 @@ flex g++ git gperf graphviz junit4 libcap-dev libcppunit-dev \
 libcppunit-doc libcunit1 libcunit1-dev libegl1-mesa-dev libfontconfig1-dev libgl1-mesa-dev \
 libgtk-3-dev libgtk2.0-dev libkrb5-dev libpcap0.8 libpcap0.8-dev libtool \
 libxml2-utils libxrandr-dev libxrender-dev libxslt1-dev libxt-dev m4 nasm openssl libssl-dev \
-pkg-config python-dev python-polib python3-dev uuid-runtime xsltproc \
+pkg-config python-dev python-polib python3-dev uuid-runtime xsltproc libcap2-bin \
   ${DIST_PKGS} -y; then
     exit 1
 fi

--- a/bin/systemSetup.sh
+++ b/bin/systemSetup.sh
@@ -23,18 +23,29 @@ apt-get install lsb-release -y
 DIST=$(lsb_release -si)
 # RELEASE=`lsb_release -sr`
 
+CODENAME=$(lsb_release -c)
+CODENAME=`echo ${CODENAME##*:}`
+
 DIST_PKGS=""
 if [ "${DIST}" = "Ubuntu" ]; then
   DIST_PKGS="${DIST_PKGS} openjdk-8-jdk"
 fi
 if [ "${DIST}" = "Debian" ]; then
-  DIST_PKGS="${DIST_PKGS} openjdk-7-jdk"
+  if [ "${CODENAME}" = "stretch" ];then
+    DIST_PKGS="${DIST_PKGS} openjdk-8-jdk"
+    DIST_PKGS="${DIST_PKGS} libpng16.16"
+    DIST_PKGS="${DIST_PKGS} libpng-dev"
+  else
+    DIST_PKGS="${DIST_PKGS} openjdk-7-jdk"
+    DIST_PKGS="${DIST_PKGS} libpng12-0"
+    DIST_PKGS="${DIST_PKGS} libpng12-dev"
+  fi
 fi
 
 if ! apt-get install ant sudo systemd wget zip make procps automake bison ccache \
 flex g++ git gperf graphviz junit4 libcap-dev libcppunit-dev \
 libcppunit-doc libcunit1 libcunit1-dev libegl1-mesa-dev libfontconfig1-dev libgl1-mesa-dev \
-libgtk-3-dev libgtk2.0-dev libkrb5-dev libpcap0.8 libpcap0.8-dev libpng12-0 libpng12-dev libtool \
+libgtk-3-dev libgtk2.0-dev libkrb5-dev libpcap0.8 libpcap0.8-dev libtool \
 libxml2-utils libxrandr-dev libxrender-dev libxslt1-dev libxt-dev m4 nasm openssl \
 pkg-config python-dev python-polib python3-dev uuid-runtime xsltproc \
   ${DIST_PKGS} -y; then

